radixdlt/babylon-wallet-android
# [ABW-3836] - Invalid transaction handling improvements

[ABW-3836](https://radixdlt.atlassian.net/browse/ABW-3836)

## Description
When a transaction is received, a request for the transaction preview is made to the GW. If the preview request returns an error, the transaction is deemed invalid. In this case, updated the UI to hide the irrelevant UI elements.

## Main changes
-  Refactored`PeerdroidClient.pareseIncomingMessage` method to separate the exception handling based on the scenario:
   - Ledger response handling;
   - dApp interaction version validation;
   - dApp response handling;
- Moved the incoming requests handling logic from the `MainViewModel` to a delegate.

## How to test
1. Launch a transaction with an invalid manifest
2. Check the error pop up
3. Tap Ok and check that the tx review screen is dismissed

## PR submission checklist


[ABW-3836]: https://radixdlt.atlassian.net/browse/ABW-3836?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
